### PR TITLE
Add Vagrant config for python3 dev environment

### PR DESCRIPTION
**Pull request by thoughts1053**
This sets up the environment for local development. It configures a Xenial (16.04 LTS) Ubuntu box with Git (including your credentials), Python 3, and Pip3.

Only difference with python 3 is you'll need to use the command `python3` on the command line instead of the default `python`.

It also includes the commands for when we have a Flask app and python requirements file. Lines 39 and 40 will just need to be uncommented.

### CODE REVIEW

**Comment by radhikamattoo** [comment]
Do we wanna merge my PL first (with requirements.txt) then comment out the lines in the vagrant file and merge yours?

**Comment by thoughts1053** [comment]
@radhikamattoo Yeah, sounds good. Just commented on your PR.

**Comment by carpetri** [comment]
@thoughts1053 any reason to use python3?

**Comment by thoughts1053** [comment]
@carpetri I went with python3 because it's the current version of the language and version 2 isn't getting any new major releases. Plus it also comes on Xenial by default.

https://wiki.python.org/moin/Python2orPython3
TLDR: Python 2.x is legacy, Python 3.x is the present and future of the language

**Comment by radhikamattoo** [comment]
hey yall i just pulled from master and did `vagrant up`, `vagrant ssh`, then `cd /vagrant && python3 server.py` and it turns out the requirements.txt file isn’t installed even though it’s in the vagrantfile.  Here’s my terminal output (ignore the fact that I forget about python3 lol): 

screen shot 2017-10-02 at 5 20 03 pm

**Comment by carpetri** [comment]
That's must've  been because you previously did the `vagrant up` then the merge came so now you just need to do `vagrant provision` before going into the vm

**Comment by thoughts1053** [comment]
@radhikamattoo just retested with master and its working. If `vagrant provision` doesn't work for you, run `vagrant destroy` and then `vagrant up` again.

**Comment by radhikamattoo** [comment]
Word, we're all good - I totally forgot about `vagrant provision` tbh
